lib/rules.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Rules do
+  def is_won(board, marker) do
+    board
+    |> get_combos()
+    |> check_combos(marker)
+  end
+
+  def is_draw(board, marker) do
+    Board.is_filled(board) and not is_won(board, marker)
+  end
+
+  def is_over(board, marker) do
+    is_draw(board, marker) or is_won(board, marker)
+  end
+
+  def is_winning_combo(cells, marker) do
+    Enum.all?(cells, fn cell -> cell == marker end)
+  end
+
+  def get_combo_length(board) do
+    round(:math.sqrt(Kernel.length(board)))
+  end
+
+  def check_combos(sectioned_board, marker) do
+    Enum.any?(sectioned_board, fn cells -> is_winning_combo(cells, marker) end)
+  end
+
+  def get_combos(board) do
+    combo_length = get_combo_length(board)
+
+    Formatter.split_into_rows(board) ++
+      get_columns(board, combo_length) ++
```

Comment:
Should `get_columns` and `get_diagonals` be in the Formatter module as well?

Reply:
Ahh yes, I agree they should!